### ISSUE TITLE
When the dispatcher is stopped, new calls shouldn't block

### ISSUE THREAD

**Issue by jzila** [keybase/go-framed-msgpack-rpc#47]
[Here](https://github.com/keybase/go-framed-msgpack-rpc/blob/5f85a2a031aa3a8ebd07895a3576fc9f30ea774a/dispatch.go#L166), we will block if the call loop has been stopped.

This will fix it:
```go
select {
    case <-d.stopCh:
        return io.EOF
    case d.callCh <- call:
        return <-call.resultCh
}
```

**Comment by maxtaco** [comment]
This unfortunately didn't fix my bug when i tried it.

**Comment by jzila** [comment]
@maxtaco true, but it's needed anyway.

**Comment by maxtaco** [comment]
Cool.  I will write a small failing test case tonight and will see if I can
make any progress debugging it. Too much spam in my current situation.

**Comment by maxtaco** [comment]
Sorry for the bad report earlier, this *does* fix my bug.  A few nits though:

Shouldn't it be `DisconnectedError` rather than `io.EOF`?

Also, we don't get debug log messages in these cases, which I think would be useful.

**Comment by jzila** [comment]
This appears fixed, likely as a result of the fix for #47
